analyzers/src/SonarAnalyzer.CSharp/Rules/GenericTypeParametersRequired.cs

```diff
@@ -31,35 +31,50 @@ public sealed class GenericTypeParametersRequired : SonarDiagnosticAnalyzer
 
         protected override void Initialize(SonarAnalysisContext context) =>
             context.RegisterSyntaxNodeActionInNonGenerated(c =>
+            {
+                var methodDeclaration = (MethodDeclarationSyntax)c.Node;
+
+                var typeParameters = methodDeclaration
+                        .TypeParameterList
+                        ?.Parameters
+                        .Select(p => c.SemanticModel.GetDeclaredSymbol(p));
```

Comment:
general remark about what I said on figuring out a way to do some AST-based checks for name equivalence before making any call to the SemanticModel: it might not be worth the effort and additional complexity, because this code will not be executed many times, but only on generic method definitions